apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -565,7 +568,24 @@ def __init__(
         store_param_remainders: bool = False,
         with_scaled_states: bool = False,
         nccl_ub: bool = False,
+        capturable: bool = False,
     ):
+        if (with_scaled_states or store_param_remainders) and capturable:
+            raise Exception(f"{self.__class__.__name__} with scaled states "
+                "or storing param remainders doesn't support CUDA graph yet.")
+
+        if capturable and not _FOUND_DEPRECATED_FUSED_ADAM:
+            raise Exception(f"Capturable {self.__class__.__name__} relies on "
+                "multi_tensor_copy to set dummy_overflow_buf to indicate "
+                "whether there's gradient Inf/NaN, build APEX with "
+                "`--deprecated_fused_adam` is essential.")
+
+        # If capturable for CUDA graph
+        self.capturable: bool = capturable
+        # If the optimizer is capturable then LR should be a tensor (on GPU)
+        lr: torch.Tensor | float = torch.tensor(lr, dtype=torch.float32) \
+            if capturable else lr
```

Comment:
is `lr` really on cuda device with this initialization?

Reply:
Also, union typing with `|` is only supported in Python 3.10 (https://peps.python.org/pep-0604/):
```suggestion
        if capturable:
            lr = torch.tensor([lr], dtype=torch.float32, device=device)
```

Reply:
Hmm, `torch.tensor([lr])` is 1-dim tensor, ` torch.tensor(lr)` is 0-dim tensor. PyTorch LR scheduler expects the later one.